scripts/bump_deps.lua

```diff
@@ -0,0 +1,378 @@
+--
+--    # create branch
+--    nvim -es +"lua require('scripts.bump_deps').create_branch()"
+
+local M = {}
+
+local _trace = false
+local required_branch_prefix = "bump-"
+local commit_prefix = "build(deps): "
+
+-- Print message
+local function p(s)
+	vim.cmd("set verbose=1")
+	vim.api.nvim_echo({ { s, "" } }, false, {})
+	vim.cmd("set verbose=0")
+end
+
+local function die()
+	p("")
+	vim.cmd("cquit 1")
+end
+
+-- Executes and returns the output of `cmd`, or nil on failure.
+-- if die_on_fail is true, process dies with die_msg on failure
+--
+-- Prints `cmd` if `trace` is enabled.
+local function _run(cmd, die_on_fail, die_msg)
+	if _trace then
+		p("run: " .. vim.inspect(cmd))
+	end
+	local rv = vim.trim(vim.fn.system(cmd)) or ""
+	if vim.v.shell_error ~= 0 then
+		if die_on_fail then
+			p(rv)
+			p(die_msg)
+			die()
+		end
+		return nil
+	end
+	return rv
+end
+
+-- Run a command, return nil on failure
+local function run(cmd)
+	return _run(cmd, false, "")
+end
+
+-- Run a command, die on failure with err_msg
+local function run_die(cmd, err_msg)
+	return _run(cmd, true, err_msg)
+end
+
+local function check_executable(cmd)
+	local cmd_path = run({ "command", "-v", cmd })
+	if cmd_path == nil then
+		return false
+	end
+	local rv = run({ "test", "-x", cmd_path })
+	return rv ~= nil
+end
+
+local function require_executable(cmd)
+	local cmd_path = run_die({ "command", "-v", cmd }, cmd .. " not found!")
+	run_die({ "test", "-x", cmd_path }, cmd .. " is not executable")
+end
+
+local function rm_file_if_present(path_to_file)
+	run({ "rm", "-f", path_to_file })
+end
+
+local nvim_src_dir = vim.fn.getcwd()
+local temp_dir = nvim_src_dir .. "/tmp"
+run({ "mkdir", "-p", temp_dir })
+local gh_res_path = temp_dir .. "/gh_res.json"
+
+local function get_dependency(dependency_name)
+	local dependency_table = {
+		["LuaJIT"] = {
+			repo = "LuaJIT/LuaJIT",
+			symbol = "LUAJIT",
+		},
+		["libuv"] = {
+			repo = "libuv/libuv",
+			symbol = "LIBUV",
+		},
+		["Luv"] = {
+			repo = "luvit/luv",
+			symbol = "LUV",
+		},
+		["tree-sitter"] = {
+			repo = "tree-sitter/tree-sitter",
+			symbol = "TREESITTER",
+		},
+	}
+	local dependency = dependency_table[dependency_name]
+	if dependency == nil then
+		p("Not a dependency: " .. dependency_name)
+		die()
+	end
+	dependency.name = dependency_name
+	return dependency
+end
+
+local function dl_gh_ref_info(repo, ref)
+	require_executable("curl")
+
+	rm_file_if_present(gh_res_path)
+	local ref_status_code = run_die({
+		"curl",
+		"-s",
+		"-H",
+		"'Accept: application/vnd.github.v3+json'",
+		"-w",
+		"'%{http_code}'",
+		"https://api.github.com/repos/" .. repo .. "/commits/" .. ref,
+		"-o",
+		gh_res_path,
+	}, "Failed to fetch commit details from GitHub")
+	if ref_status_code ~= "'200'" then
+		p("Not a valid ref: " .. ref)
+		die()
+	end
+end
+
+local function get_json_field(jsonfile_path, field)
+	require_executable("jq")
+	return run({ "jq", "-r", "." .. field, jsonfile_path })
+end
```

Comment:
You can use `gh api` directly without requiring `jq` explicitly, while still being able to run `jq` queries

```shell
gh api repos/libuv/libuv/releases --jq '.[1]'
```

Reply:
Done. Thanks, this is way cleaner!